public/js/app.Service.js

```diff
@@ -0,0 +1,90 @@
+App.service('AppService', function() {
+    const SEQUENCE_LENGTH = 20;
+    var sequence = [];
+    var userInput = [];
+    var j = 0;
+
+    this.getSequenceLength = function() {
+        return SEQUENCE_LENGTH;
+    };
+
+    this.generateSequence = function(n) {
+        for(var i = 0; i < n; i++)
+            sequence.push(Math.floor(Math.random()*4) + 1);
+    };
+
+    this.getColorSequence = function() {
+        var colorObjs = [];
+        for(var i = 0; i < SEQUENCE_LENGTH; i++) {
+            colorObjs.push(this.getColor(i));
+        }
+        return colorObjs;
+    };
+
+    this.resetSequence = function() {
+        sequence = [];
+    };
+
+    this.getColor = function(i) {
+        var colorCode = sequence[i];
+        return {code: colorCode,
+                lightColorObj: this.getLightColorObj(colorCode),
+                darkColorObj: this.getDarkColorObj(colorCode)
+            }
+    };
+
```

Comment:
Definitely leverage angular directives to handle this logic for you.  Its easy in angular ngClass and ngClick to say if I do this then change this scope value, and if this scope value == something then toggle this css class.